integration-tests/tests/janitor.js

```diff
@@ -0,0 +1,44 @@
+const rawConsole = require('console')
+const dateFns = require('date-fns/fp')
+const _ = require('lodash/fp')
+const { testWorkspaceNamePrefix } = require('../utils/integration-helpers')
+const { dismissNotifications, signIntoTerra } = require('../utils/integration-utils')
+const { withUserToken } = require('../utils/terra-sa-utils')
+
+
+const olderThanDays = 2
+
+const runJanitor = withUserToken(async ({ billingProject, page, testUrl, token }) => {
+  // Sign into Terra so we have the correct credentials.
+  await page.goto(testUrl)
+  await signIntoTerra(page, token)
+  await dismissNotifications(page)
+
+  // Delete old orphaned workspaces. These can result from local integration test runs where the
+  // process was prematurely terminated, but a few also appear in alpha and staging indicating that
+  // ci test runs also sometimes leak workspaces.
+  const workspaces = await page.evaluate(async () => await window.Ajax().Workspaces.list())
+  const oldWorkspaces = _.filter(({ workspace: { namespace, name, createdDate } }) => {
+    const age = dateFns.differenceInDays(new Date(createdDate), new Date())
+    // rawConsole.info(`${namespace} ${name}, age ${age} days`)
+    return namespace === billingProject && _.startsWith(testWorkspaceNamePrefix, name) && age > olderThanDays
+  }, workspaces)
+
+  rawConsole.log(`Deleting ${oldWorkspaces.length} workspaces with prefix "${testWorkspaceNamePrefix}" created more than ${olderThanDays} days ago.`)
+
+  return Promise.all(_.map(async ({ workspace: { namespace, name } }) => {
```

Comment:
@nawatts unfortunately there's no batch delete API endpoint.

Reply:
I meant like making the individual requests in batches of a limited size vs sending them all at once. Though I guess that since the test runs regularly, there's unlikely to be a very large number of workspaces that need to be deleted.

Reply:
I don't think there's any harm in sending them all at once. As you noted, the number is unlikely to ever be very large since we run it often (the number of workspaces we are leaking per day in dev is in the range of 3-30). I did run this to delete the old workspaces, and then I was deleting 100s at a time. I hit the 404 issue from sam that I addressed in the deleteWorkspace endpoint, but otherwise there weren't problems. And it was actually good that it revealed the 404 issue, which had caused very sporadic failures both on swatomation tests and integration tests in the past.